timemachine/cpp/src/kernels/k_hilbert.cu

```diff
@@ -12,41 +14,43 @@ void __global__ k_coords_to_kv_gather(
     const unsigned int *__restrict__ bin_to_idx,
     unsigned int *__restrict__ keys,
     unsigned int *__restrict__ vals) {
+    int idx = blockIdx.x * blockDim.x + threadIdx.x;
 
-    const int idx = blockIdx.x * blockDim.x + threadIdx.x;
+    const double bx = box[0 * 3 + 0];
+    const double by = box[1 * 3 + 1];
+    const double bz = box[2 * 3 + 2];
 
-    if (idx >= N) {
-        return;
-    }
+    const double inv_bx = 1 / bx;
+    const double inv_by = 1 / by;
+    const double inv_bz = 1 / bz;
```

Comment:
To confirm, is the fix for the out-of-bounds memory access the change to double precision here?

Reply:
Correct, switching to double precision resolves the out of bounds memory access